datafusion/core/src/datasource/schema_adapter.rs

```diff
@@ -75,9 +75,16 @@ pub trait SchemaAdapter: Send + Sync {
 
 /// Creates a `SchemaMapping` that can be used to cast or map the columns
 /// from the file schema to the table schema.
-pub trait SchemaMapper: Send + Sync {
+pub trait SchemaMapper: Debug + Send + Sync {
     /// Adapts a `RecordBatch` to match the `table_schema` using the stored mapping and conversions.
     fn map_batch(&self, batch: RecordBatch) -> datafusion_common::Result<RecordBatch>;
+
+    /// Adapts a `RecordBatch` that does not have all the columns (as defined in the schema).
+    /// This method is slower than `map_batch` and should only be used when explicitly needed.
```

Comment:
I think it would help here to give some context about why this is slower. Here is a suggestion

```diff
-    /// Adapts a `RecordBatch` that does not have all the columns (as defined in the schema).
-    /// This method is slower than `map_batch` and should only be used when explicitly needed.
+    /// Adapts a [`RecordBatch`] that does not  have all the columns from the
+    /// file schema.
+    ///
+    /// This method is used when applying a filter to a subset of the columns during
+    /// via an `ArrowPredicate`.
+    ///
+    /// This method is slower than `map_batch` as it looks up columns by name.```